Caused by: java.lang.ClassCastException: android.os.Parcelable[] cannot be cast to com.example.bikini_android.repository.feed.Feed[] 이슈 수정

FATAL EXCEPTION: main
Process: com.example.bikini_android.debug, PID: 11998
java.lang.RuntimeException: Unable to start activity ComponentInfo{com.example.bikini_android.debug/com.example.bikini_android.ui.holder.MainHolderActivity}: java.lang.ClassCastException: android.os.Parcelable[] cannot be cast to com.example.bikini_android.repository.feed.Feed[]
at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3819)
at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3995)
at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85)
at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2309)
at android.os.Handler.dispatchMessage(Handler.java:106)
at android.os.Looper.loop(Looper.java:246)
at android.app.ActivityThread.main(ActivityThread.java:8587)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:602)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1130)
Caused by: java.lang.ClassCastException: android.os.Parcelable[] cannot be cast to com.example.bikini_android.repository.feed.Feed[]
at com.example.bikini_android.ui.feeds.FeedsFragment.onCreate(FeedsFragment.kt:57)
at androidx.fragment.app.Fragment.performCreate(Fragment.java:2949)
at androidx.fragment.app.FragmentStateManager.create(FragmentStateManager.java:475)
at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:278)
at androidx.fragment.app.FragmentStore.moveToExpectedState(FragmentStore.java:120)
at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1647)
at androidx.fragment.app.FragmentManager.dispatchStateChange(FragmentManager.java:3126)
at androidx.fragment.app.FragmentManager.dispatchCreate(FragmentManager.java:3059)
at androidx.fragment.app.Fragment.restoreChildFragmentState(Fragment.java:1891)
at androidx.fragment.app.Fragment.onCreate(Fragment.java:1867)
at androidx.navigation.fragment.NavHostFragment.onCreate(NavHostFragment.java:264)
at androidx.fragment.app.Fragment.performCreate(Fragment.java:2949)
at androidx.fragment.app.FragmentStateManager.create(FragmentStateManager.java:475)
at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:278)
at androidx.fragment.app.FragmentStore.moveToExpectedState(FragmentStore.java:120)
at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1647)
at androidx.fragment.app.FragmentManager.dispatchStateChange(FragmentManager.java:3126)
at androidx.fragment.app.FragmentManager.dispatchCreate(FragmentManager.java:3059)
at androidx.fragment.app.FragmentController.dispatchCreate(FragmentController.java:240)
at androidx.fragment.app.FragmentActivity.onCreate(FragmentActivity.java:276)
at androidx.appcompat.app.AppCompatActivity.onCreate(AppCompatActivity.java:115)
at com.example.bikini_android.ui.holder.MainHolderActivity.onCreate(MainHolderActivity.kt:35)
at android.app.Activity.performCreate(Activity.java:8207)
at android.app.Activity.performCreate(Activity.java:8191)
at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1309)
at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3792)
... 11 more

--- a/app/src/main/java/com/example/bikini_android/ui/feeds/FeedsFragment.kt
+++ b/app/src/main/java/com/example/bikini_android/ui/feeds/FeedsFragment.kt
@@ -53,8 +53,16 @@ class FeedsFragment : BaseFragment() {
                 it.getString(KEY_SORT_TYPE_NAME) ?: FeedsSortType.NEAR_DISTANCE.name
             )
 
-            @Suppress("UNCHECKED_CAST")
-            feedsReceived = (it.getParcelableArray(KEY_FEEDS) as Array<Feed>?)?.toList()
+            val feedsParcelableArray = it.getParcelableArray(KEY_FEEDS)
+            feedsParcelableArray?.let {
+                val tempFeeds = mutableListOf<Feed>()
+                feedsParcelableArray.forEach { feedParcelable ->
+                    (feedParcelable as? Feed)?.let { feed ->
+                        tempFeeds.add(feed)
+                    }
+                }
+                feedsReceived = tempFeeds
+            }
         }
     }
 
@@ -110,7 +118,8 @@ class FeedsFragment : BaseFragment() {
                         RecyclerViewLayoutType.VERTICAL,
                         feedsType,
                         sortType,
-                        event.feed
+                        event.feed,
+                        feedsReceived
                     )
                 )
             }.addTo(disposables)

